src/event/sys/unix/parse.rs

```diff
@@ -170,6 +170,13 @@ pub(crate) fn parse_csi(buffer: &[u8]) -> Result<Option<InternalEvent>> {
         b'I' => Some(Event::FocusGained),
         b'O' => Some(Event::FocusLost),
         b';' => return parse_csi_modifier_key_code(buffer),
+        // P, Q, and S for compatibility with Kitty keyboard protocol,
+        // as the 1 in 'CSI 1 P' etc. must be omitted if there are no
+        // modifiers pressed:
+        // https://sw.kovidgoyal.net/kitty/keyboard-protocol/#legacy-functional-keys
+        b'P' => Some(Event::Key(KeyCode::F(1).into())),
```

Comment:
Wonder if this could cause compatibility issues with nonkitty sequences. Are we sure this pattern never occurs when kitty is not enabled?

Reply:
There aren't any conflicts in [the xterm documentation](https://www.x.org/docs/xterm/ctlseqs.pdf) for sequences sent from the terminal starting with CSI and immediately terminated by P, Q, or S.

The closest is CSI ps P with ps having a default value of 1 (so it can be omitted, shortening the sequence to CSI P), but that is a sequence sent *to* the terminal from the program, not the other way around.

Reply:
Alright, then it is fine. If in future it becomes an issue we likely can split it out and use [this pr ](https://github.com/crossterm-rs/crossterm/pull/732)changes to check if kitty is enabled and check for those sequences.